latch_cli/centromere/ctx.py

```diff
@@ -223,28 +238,98 @@ def full_image(self):
         """
         return f"{self.dkr_repo}/{self.image}"
 
-    def nucleus_provision_url(self) -> Tuple[str, str]:
+    def provision_register_deployment(self) -> Tuple[str, str]:
         """Retrieve centromere IP + username."""
+        print("Provisioning register instance. This may take a few minutes.")
 
-        headers = {"Authorization": f"Bearer {self.token}"}
+        resp = tinyrequests.post(
+            "https://centromere.latch.bio/register/start",
+            headers={"Authorization": f"Latch-SDK-Token {self.token}"},
+            json={"SSHKey": self.ssh_key_path.with_suffix(".pub").read_text()},
+        )
 
-        response = tinyrequests.post(
-            self.latch_provision_url,
-            headers=headers,
-            json={
-                "public_key": self.public_key,
-            },
+        json_data = resp.json()
+        if resp.status_code != 200:
+            raise ValueError(json_data["Error"])
+
+        ip = json_data["IP"]
+        self.jump_key_path.write_text(json_data["JumpKey"])
+        self.jump_key_path.chmod(0o600)
+
+        subprocess.run(
+            ["ssh-add", str(self.jump_key_path)],
```

Comment:
we really should get rid of `ssh-add`, could specify the ssh key file directly in the config instead basically

Reply:
yeah i can do that next week